crates/flux-middle/src/queries.rs

```diff
@@ -536,6 +542,27 @@ impl<'genv, 'tcx> Queries<'genv, 'tcx> {
         })
     }
 
+    pub(crate) fn default_assoc_refinement_def(
+        &self,
+        genv: GlobalEnv,
+        trait_id: DefId,
+        name: Symbol,
+    ) -> QueryResult<Option<rty::EarlyBinder<rty::Lambda>>> {
+        run_with_cache(&self.default_assoc_refinement_def, (trait_id, name), || {
+            let trait_id = lookup_extern(genv, trait_id).unwrap_or(trait_id);
+            if let Some(local_id) = trait_id.as_local() {
+                (self.providers.default_assoc_refinement_def)(genv, local_id, name)
+            } else if let Some(lam) = genv
+                .cstore()
+                .default_assoc_refinements_def((trait_id, name))
+            {
+                lam
+            } else {
+                bug!("TODO: implement for external crates")
```

Comment:
Prefer `query_bug!` if it's easy to return a `QueryResult`.

```suggestion
                Err(query_bug!("TODO: implement for external crates"))
```

Also, I think this is truly unreachable rather than a TODO.